### ISSUE TITLE
Only one GPU is used in the training for a mult-GPU machine with mirrored_strategy

### ISSUE THREAD

**Issue by amir1m** [amaiya/ktrain#459]
@amaiya 
As per the example in https://github.com/amaiya/ktrain/issues/78, I am trying to train the NER model on custom CONLL dataset(5.5million rows):

- Number of sentences:  228051
- Number of words in the dataset:  232634
- Number of Labels:  29
- Longest sentence: 106 words

Model
```
with mirrored_strategy.scope():
    model = txt.sequence_tagger('bilstm-transformer', preproc, wv_path_or_url='/home/user1/ktrain_data/cc.en.300.vec')
learner = ktrain.get_learner(model, train_data=trn, val_data=val, batch_size=128)
learner.fit(0.003, 5, cycle_len=1)
```

Only 1st GPU seems to be used as reported by `nvidia-smi` utilizing most of the times 100%. Other GPUs gets consumer in single digit if they ever are consumed. What is missing? How do take advantage of all the GPUs?

**Comment by amaiya** [comment]
Are you doing things in the way shown in  [this FAQ entry](https://github.com/amaiya/ktrain/blob/master/FAQ.md#how-do-i-train-using-multiple-gpus) and setting `os.environ["CUDA_VISIBLE_DEVICES"]` correctly?

**Comment by amir1m** [comment]
Yes, those environment variables are exported as in the example.  
 
  On Mon, 22 Aug 2022 at 3:28, Arun S. ***@***.***> wrote:   


Are you doing things in the way shown in this FAQ entry and setting os.environ["CUDA_VISIBLE_DEVICES"] correctly?

**Comment by amir1m** [comment]
@amaiya , Also CRF  variants of sequence taggers do not work even after enabling TF1 compatibility mode as indicated in error/warning message.

**Comment by amaiya** [comment]
The `bilstm-crf` model works for me using TensorFlow 2.9 and Python 3.9.  You can post/share a reproducible example of the problem on Google Colab, if you need further help.

With regard to the multi-gpu issue, I will try looking into this next week after our local  multi-GPU machines are updated and will keep this issue open for now.

**Comment by amaiya** [comment]
The `transformer` embeddings with BiLSTM keras model may be causing the multi-gpu issue.  I don't have a fix for this unfortunately, but would welcome PRs if you do.
